salt/returners/couchdb_return.py

```diff
@@ -147,22 +220,47 @@ def returner(ret):
     options = _get_options(ret)
 
     # Check to see if the database exists.
-    _response = _request("GET", options['url'] + "_all_dbs")
+    _response = _request("GET",
+                         options['url'] + "_all_dbs",
+                         user=options['user'],
+                         passwd=options['passwd'])
     if options['db'] not in _response:
 
         # Make a PUT request to create the database.
-        _response = _request("PUT", options['url'] + options['db'])
+        _response = _request("PUT",
+                             options['url'] + options['db'],
+                             user=options['user'],
+                             passwd=options['passwd'])
 
         # Confirm that the response back was simple 'ok': true.
         if 'ok' not in _response or _response['ok'] is not True:
-            log.error('Unable to create database \'%s\'', options['db'])
-            log.error('Nothing logged! Lost data.')
+            log.error('Nothing logged! Lost data. Unable to create database "{0}"'.format(options['db']))
+            log.debug('_response object is: {0}'.format(_response))
             return
-        log.info('Created database \'%s\'', options['db'])
-
-    # Call _generate_doc to get a dict object of the document we're going to
-    # shove into the database.
-    doc = _generate_doc(ret)
+        log.info('Created database "{0}"'.format(options['db']))
+
+    if boltons_lib:
+        # redact all passwords if options['redact_pws'] is True
+        if options['redact_pws']:
+            log.info('Redacting passwords.')
+            ret_remap_pws = remap(ret, visit=_redact_passwords)
+        else:
+            log.info('Not redacting passwords.')
+            ret_remap_pws = ret
+
+        # remove all return values starting with '__pub' if options['minimum_return'] is True
+        if options['minimum_return']:
+            log.info('Minimizing the return data.')
+            ret_remapped = remap(ret_remap_pws, visit=_minimize_return)
+        else:
+            log.info('Not minimizing the return data.')
+            ret_remapped = ret_remap_pws
+    else:
+        log.info('boltons library not installed. Cannot redact passwords or minimize the return')
```

Comment:
Could we please include some info here about where this library can be found and how it can be installed?